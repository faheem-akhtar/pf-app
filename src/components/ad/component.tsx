import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
