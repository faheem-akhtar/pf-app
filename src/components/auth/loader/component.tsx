import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
