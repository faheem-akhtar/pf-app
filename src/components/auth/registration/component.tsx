import { useState } from 'react';
import { useTranslationHook } from 'helpers/hook/translation.hook';

import { ApiFetcherResultFailureInterface } from 'api/fetcher-result-failure.interface';
import { AuthRegisterService } from 'services/auth/register.service';
import { AuthRegistrationFieldEnum } from 'components/auth/registration/field.enum';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { CheckboxTemplate } from 'library/checkbox/template';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

const captchaService = new GoogleRecaptchaService();

export const AuthRegistrationComponent = ({ close }: { close: () => void }): JSX.Element => {
  const { t } = useTranslationHook();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [optedIn, setOptedIn] = useState(false);
  const [errors, setErrors] = useState<Partial<Record<AuthRegistrationFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthRegistrationFieldEnum.email]: [
      validationRequired(t('Enter your email')),
      validationEmail(t('Please enter a valid email')),
    ],
    [AuthRegistrationFieldEnum.password]: [validationRequired(t('Please enter a password'))],
    [AuthRegistrationFieldEnum.firstName]: [validationRequired(t('Please enter a valid first and last name'))],
    [AuthRegistrationFieldEnum.lastName]: [validationRequired(t('Please enter a valid first and last name'))],
  }));

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(setErrors, validators);

  // TODO-FE[CX-219] - Update template
  return (
    <>
      {errorMessage && <div className='authentication__error--general'>{errorMessage}</div>}
      <form
        className='authentication__form'
        onSubmit={(e): void => {
          e.preventDefault();
          if (
            validate({
              [AuthRegistrationFieldEnum.email]: email,
              [AuthRegistrationFieldEnum.password]: password,
              [AuthRegistrationFieldEnum.firstName]: firstName,
              [AuthRegistrationFieldEnum.lastName]: lastName,
            })
          ) {
            captchaService.execute().then((captcha_token) => {
              AuthRegisterService({
                email,
                password,
                first_name: firstName,
                last_name: lastName,
                opted_in: optedIn,
                captcha_token,
              }).then((e) => {
                if (!(e as ApiFetcherResultFailureInterface).ok) {
                  setErrorMessage(
                    (e as ApiFetcherResultFailureInterface).error.body ||
                      t('Something went wrong! Please try again later')
                  );
                  captchaService.reset();
                } else {
                  // Reset server error message
                  setErrorMessage('');
                  // Close modal
                  close();
                }
              });
            });
          }
          return;
        }}
      >
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your email'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.email]: value,
              });
              setEmail(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.email]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your first name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.firstName]: value,
              });
              setFirstName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.firstName]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your last name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.lastName]: value,
              });
              setLastName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.lastName]}</div>
        </div>
        <div className='authentication__input-area'>
          <input
            type='password'
            className='authentication__input'
            value={password}
            onChange={(e): void => {
              const value = e.target.value;
              !validate({
                [AuthRegistrationFieldEnum.password]: value,
              });
              setPassword(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.password]}</div>
        </div>
        <div className='authentication__input-area'>
          <CheckboxTemplate
            id='Opten in'
            checked={optedIn}
            onChange={(e): void => {
              setOptedIn(!!e.target.value);
            }}
          >
            {t('filters-modal/commercial-only')}
          </CheckboxTemplate>
        </div>

        <div className='authentication__form-submit'>
          <ButtonTemplate type='submit' componentType={ButtonComponentTypeEnum.tertiary} size={ButtonSizeEnum.small}>
            {'Register'}
          </ButtonTemplate>
        </div>
      </form>
    </>
  );
};
