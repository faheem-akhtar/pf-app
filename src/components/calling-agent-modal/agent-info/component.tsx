import Image from 'next/image';
import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './calling-agent-modal-agent-info.module.scss';
import { CallingAgentModalAgentInfoComponentPropsInterface } from './component-props.interface';

export const CallingAgentModalAgentInfoComponent: FunctionComponent<CallingAgentModalAgentInfoComponentPropsInterface> =
  ({ referenceId, imageSrc, languages, name }) => {
    const { t } = useTranslation();

    return (
      <div className={styles.container}>
        <div className={styles['image-container']}>
          {imageSrc ? (
            <picture>
              <source srcSet={imageSrc} type='image/jpeg' />
              <img className={styles.image} src={imageSrc} alt={name} />
            </picture>
          ) : (
            <Image
              src='/static/images/agent-image-placeholder.jpg'
              width={70}
              height={70}
              objectFit='cover'
              alt='Picture of the agent'
            />
          )}
        </div>
        <div className={styles.content}>
          <p className={styles.name}>{name}</p>
          <p className={styles.description}>
            {languages?.length ? (
              <>
                {`${t('speaks')}: ${languages.join(', ')}`}
                <br />
              </>
            ) : null}
            {t('reference')}: <span className={styles.ref}>{referenceId}</span>
          </p>
        </div>
      </div>
    );
  };
