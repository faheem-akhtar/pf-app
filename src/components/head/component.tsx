import Head from 'next/head';
import { useRouter } from 'next/router';

import { appleItunesAppEnabledByDefault } from 'config/apple-itunes-app/enabled-by-default';
import { configCommon } from 'config/common';
import { localeIsDefault } from 'helpers/locale/is-default';

import { HeadComponentPropsInterface } from './component-props.interface';
import { HeadTrackersTemplate } from './trackers-template';

export const HeadComponent = ({
  title,
  description,
  schema,
  shouldIndex = true,
  snowplowHost,
  pageUrl,
  alternateUrl,
  pageNextUrl,
  pagePreviousUrl,
}: HeadComponentPropsInterface): JSX.Element => {
  const { locale } = useRouter();
  const { current, alternative } = configCommon.language;
  const targetLocale = localeIsDefault(locale as string) ? alternative : current;

  return (
    <Head>
      <title>{title}</title>
      <meta property='og:title' content={title} />
      {description && (
        <>
          <meta name='description' content={description} />
          <meta property='og:description' content={description} />
        </>
      )}
      {pageUrl && (
        <>
          <link href={pageUrl} rel='canonical' />
          <link href={pageUrl} rel='alternate' hrefLang={locale} />
          <meta property='og:url' content={pageUrl} />
        </>
      )}
      <meta property='og:type' content='website' />
      <meta property='og:site_name' content='Property Finder' />
      {alternateUrl && <link href={alternateUrl} rel='alternate' hrefLang={targetLocale} />}
      {pagePreviousUrl && <link href={pagePreviousUrl} rel='prev' />}
      {pageNextUrl && <link href={pageNextUrl} rel='next' />}

      <link rel='icon' type='image/png' sizes='16x16' href='/en/static/favicon/favicon-16x16.png' />
      <link rel='icon' type='image/png' sizes='32x32' href='/en/static/favicon/favicon-32x32.png' />
      <link rel='shortcut icon' type='image/x-icon' sizes='16x16 32x32 48x48' href='/en/static/favicon/favicon.ico' />
      <link rel='icon' type='image/png' sizes='192x192' href='/en/static/favicon/android-icon-192x192.png' />
      <link rel='apple-touch-icon' href='/en/static/favicon/apple-icon-76x76.png' />
      <link rel='apple-touch-icon' sizes='76x76' href='/en/static/favicon/apple-icon-76x76.png' />
      <link rel='apple-touch-icon' sizes='120x120' href='/en/static/favicon/apple-icon-120x120.png' />
      <link rel='apple-touch-icon' sizes='152x152' href='/en/static/favicon/apple-icon-152x152.png' />
      <link rel='apple-touch-icon' sizes='180x180' href='/en/static/favicon/apple-icon-180x180.png' />
      <meta name='msapplication-TileColor' content='#ffffff' />
      <meta name='msapplication-TileImage' content='/en/static/favicon/ms-icon-144x144.png' />
      <meta name='msapplication-config' content='/en/static/favicon/browserconfig.xml' />
      <meta name='msapplication-tap-highlight' content='no' />
      <meta name='theme-color' content='#ffffff' />
      <link rel='mask-icon' href='/en/static/favicon/safari-pinned-tab.svg' color='#ef5e4e' />
      <link rel='manifest' href='/en/static/site.webmanifest' />
      <meta charSet='utf-8' />
      <meta name='google' content='notranslate' />
      <meta name='locale' content={locale} />
      <meta name='viewport' content='width=device-width, initial-scale=1' />
      <meta name='robots' content={`${shouldIndex ? '' : 'no'}index,follow`} />
      <meta name='application-name' content='Property Finder' />
      <meta name='mobile-web-app-capable' content='yes' />
      <meta name='apple-mobile-web-app-capable' content='yes' />
      <meta name='apple-mobile-web-app-title' content='Property Finder' />
      <meta name='apple-mobile-web-app-status-bar-style' content='default' />
      {/* Do not render telephone numbers as links */}
      <meta name='format-detection' content='telephone=no' />
      {appleItunesAppEnabledByDefault && (
        <meta name='apple-itunes-app' content='app-id=897540233, affiliate-data=pt=94765875&amp;ct=smart_app_banner' />
      )}
      {schema && <script type='application/ld+json' dangerouslySetInnerHTML={{ __html: schema }} />}
      {snowplowHost && <HeadTrackersTemplate snowplowHost={snowplowHost} />}
    </Head>
  );
};
