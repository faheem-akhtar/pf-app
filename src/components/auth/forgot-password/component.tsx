import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
