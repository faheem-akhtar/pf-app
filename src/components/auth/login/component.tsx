import React, { useState } from 'react';

import { AuthFacebookService } from 'services/auth/facebook.service';
import { AuthGoogleService } from 'services/auth/google.service';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthLoginPropsInterface } from 'components/auth/login/props.interface';
import { AuthLoginService } from 'services/auth/login.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';
const captchaService = new GoogleRecaptchaService();

export const AuthLoginComponent = (props: AuthLoginPropsInterface): JSX.Element => {
  const { t } = useTranslation();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [AuthLoginFieldEnum.password]: [validationRequired(t('auth/empty-password'))],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  return (
    <div>
      <div className={styles.heading}>{t('sign-in')}</div>
      {/* Facebook */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.facebook)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthFacebookService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-facebook')}
      </ButtonTemplate>
      {/* Google */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.google)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthGoogleService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-google')}
      </ButtonTemplate>
      <div className={styles.divider}>
        <span className={styles.text}>{t('auth/or')}</span>
      </div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthLoginFieldEnum.email]: email,
              [AuthLoginFieldEnum.password]: password,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthLoginService({
                email,
                password,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                  } else {
                    setErrorMessage(e.error.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthLoginFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthLoginFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthLoginFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('password')}
            type='password'
            value={password}
            error={!!errors[AuthLoginFieldEnum.password]}
            onChange={(value): void => {
              !validate({
                [AuthLoginFieldEnum.password]: value,
              });
              setPassword(value);
            }}
            errorText={errors[AuthLoginFieldEnum.password]}
          />
        </div>
        <div className={styles['input-area']} onClick={props.onForgotPassword}>
          <div className={domClassMerge(styles.link, styles['forgot-password'])}>{t('auth/forgot-password')}?</div>
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('log-in')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
