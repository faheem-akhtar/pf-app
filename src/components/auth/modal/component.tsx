import React, { useState } from 'react';

import { AuthForgotPasswordComponent } from 'components/auth/forgot-password/component';
import { AuthLoginComponent } from 'components/auth/login/component';
import { AuthRegistrationComponent } from 'components/auth/registration/component';
import { AuthScreenEnum } from 'components/auth/screen.enum';
import { IconThickCrossTemplate } from 'components/icon/thick/cross-template';
import { functionNoop } from 'helpers/function/noop';

import styles from './auth-modal-component.module.scss';
import { AuthModalPropsInterface } from './props.interface';

export const AuthModalComponent = ({
  close = functionNoop,
  cancel = functionNoop,
  success = functionNoop,
}: AuthModalPropsInterface): JSX.Element => {
  const [authScreen, setAuthScreen] = useState(AuthScreenEnum.login);
  return (
    <div
      className={styles.content}
      onClick={(e): void => {
        e.stopPropagation();
      }}
    >
      <div className={styles.header}>
        <button
          onClick={(): void => {
            close();
            cancel();
          }}
        >
          <IconThickCrossTemplate class={styles.closeIcon} />
        </button>
      </div>
      {authScreen === AuthScreenEnum.login && (
        <AuthLoginComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
          onForgotPassword={(): void => setAuthScreen(AuthScreenEnum.forgotPassword)}
        />
      )}
      {authScreen === AuthScreenEnum.registration && (
        <AuthRegistrationComponent
          onClose={close}
          onSuccess={success}
          onLogin={(): void => setAuthScreen(AuthScreenEnum.login)}
        />
      )}
      {authScreen === AuthScreenEnum.forgotPassword && (
        <AuthForgotPasswordComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
        />
      )}
    </div>
  );
};
