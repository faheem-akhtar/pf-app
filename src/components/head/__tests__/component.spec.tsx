/* eslint-disable testing-library/no-node-access */
import { render } from '@testing-library/react';

import { HeadComponent } from '../component';

jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }): React.ReactElement => {
      return <>{children}</>;
    },
  };
});

describe('HeadComponent', () => {
  it('should have common tags', async () => {
    const { container } = render(<HeadComponent title='Title' />);

    expect(container).toMatchSnapshot();
  });

  it('should have description and urls', () => {
    const pageUrl = 'example.com';
    const description = 'Description';

    render(<HeadComponent title='Title' description={description} pageUrl={pageUrl} />);

    expect(document.querySelector("meta[property='og:description']")?.attributes.getNamedItem('content')?.value).toBe(
      description
    );
    expect(document.querySelector("meta[name='description']")?.attributes.getNamedItem('content')?.value).toBe(
      description
    );
    expect(document.querySelector("meta[property='og:url']")?.attributes.getNamedItem('content')?.value).toBe(pageUrl);
    expect(document.querySelector("link[rel='alternate']")?.attributes.getNamedItem('href')?.value).toBe(pageUrl);
    expect(document.querySelector("link[rel='canonical']")?.attributes.getNamedItem('href')?.value).toBe(pageUrl);
  });

  it('should have prev and next urls', () => {
    const prevPageUrl = 'prevpageurl.com';
    const nextPageUrl = 'nextpageurl.com';

    render(<HeadComponent title='Title' pagePreviousUrl={prevPageUrl} pageNextUrl={nextPageUrl} />);

    expect(document.querySelector("link[rel='prev']")?.attributes.getNamedItem('href')?.value).toBe(prevPageUrl);
    expect(document.querySelector("link[rel='next']")?.attributes.getNamedItem('href')?.value).toBe(nextPageUrl);
  });

  it('should have alternate url', () => {
    const alternateUrl = 'alternateurl.com';

    render(<HeadComponent title='Title' alternateUrl={alternateUrl} />);

    expect(document.querySelector("link[rel='alternate']")?.attributes.getNamedItem('href')?.value).toBe(alternateUrl);
    expect(document.querySelector("link[rel='alternate']")?.attributes.getNamedItem('hrefLang')?.value).toBe('ar');
  });
});
