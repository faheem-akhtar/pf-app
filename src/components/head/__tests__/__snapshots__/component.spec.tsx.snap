// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadComponent should have common tags 1`] = `
<div>
  <title>
    Title
  </title>
  <meta
    content="Title"
    property="og:title"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="Property Finder"
    property="og:site_name"
  />
  <link
    href="/en/static/favicon/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="/en/static/favicon/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="/en/static/favicon/favicon.ico"
    rel="shortcut icon"
    sizes="16x16 32x32 48x48"
    type="image/x-icon"
  />
  <link
    href="/en/static/favicon/android-icon-192x192.png"
    rel="icon"
    sizes="192x192"
    type="image/png"
  />
  <link
    href="/en/static/favicon/apple-icon-76x76.png"
    rel="apple-touch-icon"
  />
  <link
    href="/en/static/favicon/apple-icon-76x76.png"
    rel="apple-touch-icon"
    sizes="76x76"
  />
  <link
    href="/en/static/favicon/apple-icon-120x120.png"
    rel="apple-touch-icon"
    sizes="120x120"
  />
  <link
    href="/en/static/favicon/apple-icon-152x152.png"
    rel="apple-touch-icon"
    sizes="152x152"
  />
  <link
    href="/en/static/favicon/apple-icon-180x180.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <meta
    content="#ffffff"
    name="msapplication-TileColor"
  />
  <meta
    content="/en/static/favicon/ms-icon-144x144.png"
    name="msapplication-TileImage"
  />
  <meta
    content="/en/static/favicon/browserconfig.xml"
    name="msapplication-config"
  />
  <meta
    content="no"
    name="msapplication-tap-highlight"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <link
    color="#ef5e4e"
    href="/en/static/favicon/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <link
    href="/en/static/site.webmanifest"
    rel="manifest"
  />
  <meta
    charset="utf-8"
  />
  <meta
    content="notranslate"
    name="google"
  />
  <meta
    content="en"
    name="locale"
  />
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <meta
    content="index,follow"
    name="robots"
  />
  <meta
    content="Property Finder"
    name="application-name"
  />
  <meta
    content="yes"
    name="mobile-web-app-capable"
  />
  <meta
    content="yes"
    name="apple-mobile-web-app-capable"
  />
  <meta
    content="Property Finder"
    name="apple-mobile-web-app-title"
  />
  <meta
    content="default"
    name="apple-mobile-web-app-status-bar-style"
  />
  <meta
    content="telephone=no"
    name="format-detection"
  />
</div>
`;
